optimum/habana/transformers/modeling_utils.py

```diff
@@ -722,10 +728,20 @@ def adapt_transformers_to_gaudi():
     transformers.AutoTokenizer.register(BaichuanConfig, slow_tokenizer_class=BaichuanTokenizer)
     transformers.AutoModelForCausalLM.register(BaichuanConfig, BaichuanForCausalLM)
 
-    # Register chatglm with optimization on Gaudi
-    transformers.AutoConfig.register("chatglm", ChatGLMConfig)
-    transformers.AutoTokenizer.register(ChatGLMConfig, ChatGLMTokenizer)
-    transformers.AutoModel.register(ChatGLMConfig, ChatGLMForConditionalGeneration)
-    transformers.AutoModelForCausalLM.register(ChatGLMConfig, ChatGLMForConditionalGeneration)
-    transformers.AutoModelForSeq2SeqLM.register(ChatGLMConfig, ChatGLMForConditionalGeneration)
-    transformers.AutoModelForSequenceClassification.register(ChatGLMConfig, ChatGLMForSequenceClassification)
+    if os.getenv("GLM") == "4v":
```

Comment:
@mengker33, I have a suggestion. While this approach works as you implemented it, but wouldn't it be better to create custom identifiers to register both models? For example, using `chatglm` and `chatglm_4v`.

Reply:
I have tried this method before, sadly it didn't work.  Here in https://github.com/huggingface/transformers/blob/v4.45.2/src/transformers/models/auto/configuration_auto.py#L1008,  `config_dict["model_type"]` is ["chatglm"](  https://huggingface.co/THUDM/glm-4v-9b/blob/main/config.json#:~:text=%22model_type%22%3A%20%22chatglm%22%2C) while we are registering model_type "chatglm_4v", which causes `has_local_code` to be False, and then in `resolve_trust_remote_code`, it will ask for setting `trust_remote_code` as True, otherwise it failed.